packages/web-application/src/web-application.ts

```diff
@@ -901,16 +648,20 @@ export class WebApplication extends PluggableModule {
             }
 
             try {
-                let element = getElementByXPath(xpath);
-                let evt = document.createEvent('HTMLEvents');
+                const element = getElementByXPath(xpath);
 
                 if (element) {
-                    element.focus();
                     element.value = value;
+                    const eventInit = {
+                        bubbles: true,
+                    };
+                    const inputEvent = new Event('input', eventInit);
+                    const keydownEvent = new Event('keydown', eventInit);
+                    const keyupEvent = new Event('keyup', eventInit);
```

Comment:
I suggest to add check for field type. You can't trigger keyup event on radio\checkbox\select fields.
For all text fields that are accepting the keyboard events you can use `isTextNode` method from deleted code.